providers/curation/process.js

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const { get } = require('lodash')
+
+async function work() {
+  try {
+    let message = await queue.dequeue()
+    if (!get(message, 'data.pull_request') || !get(message, 'data.action')) return
+    const pr = message.data.pull_request
+    const action = message.data.action
+    switch (action) {
+      case 'opened':
+      case 'synchronize': {
+        const curations = await curationService.getContributedCurations(pr.number, pr.head.sha)
+        await curationService.validateContributions(pr.number, pr.head.sha, curations)
+        await curationService.updateContribution(pr, curations)
+        break
+      }
+      case 'closed': {
+        await curationService.updateContribution(pr)
+        break
+      }
+    }
+    logger.info(`Handled GitHub event "${action}" for PR#${pr.number}`)
+    await queue.delete(message)
+  } catch (error) {
+    logger.error(error)
```

Comment:
What will happen to the queue message in this case?

Reply:
when you call `.dequeue()` it makes the message invisible so the next call to `.dequeue()` will get the next one
After a time interval the message becomes visible again and will be in the front of the line

the DQ count increases each time this happens, and when the DQ count exceeds 5 we delete the message